shell/browser/api/electron_api_web_contents.cc

```diff
@@ -486,9 +486,13 @@ void OnCapturePageDone(gin_helper::Promise<gfx::Image> promise,
                        const SkBitmap& bitmap) {
   auto ui_task_runner = content::GetUIThreadTaskRunner({});
   if (!ui_task_runner->RunsTasksInCurrentSequence()) {
-    ui_task_runner->PostTask(
-        FROM_HERE, base::BindOnce(&OnCapturePageDone, std::move(promise),
-                                  std::move(capture_handle), bitmap));
+    if (!ui_task_runner->PostTask(
+            FROM_HERE, base::BindOnce(&OnCapturePageDone, std::move(promise),
+                                      std::move(capture_handle), bitmap))) {
+      // When `PostTask` returns `false` the task definitely will not run.
+      promise.RejectWithErrorMessage("Failed to capture page");
```

Comment:
I am sorry, just a few thoughts:

is it legitimate to reject the promise here? If `ui_task_runner->RunsTasksInCurrentSequence()` is `false` we execute the code not inside the UI thread. But as I understand, we can only resolve / reject the promise when we are in UI thread, otherwise we may encounter a crash. 
Looks like this should not be merged. cc @codebytere @zcbenz